src/main/java/com/zoo/species/Penguin.java

```diff
@@ -1,4 +1,27 @@
 package com.zoo.species;
 
-public class Penguin {
-}
+import com.zoo.animals.Bird;
+
+public class Penguin extends Bird {
+
+    public Penguin(int energyLevel, String name, int age, int weight) {
+        super(energyLevel, name, age, weight);
+    }
+
+    @Override
+    public void makeSound() {
+        System.out.println("Кря-кря!");
+    }
+
+    public void swim() {
+        System.out.println("Буль-буль");
+    }
+
+    @Override
+    public void fly() {
+        int currentEnergy = getEnergyLevel();
+        setEnergyLevel(currentEnergy - 20);
+        System.out.println(getName() + " Пінгвін не літає, але плаває:");
+        swim();
```

Comment:
I'd say if there's a private method `swim()` that actually does the job for a penguin - reduction of energy would happen inside the `swim()` method as currently there's no direct relation between how penguin swims and how much energy does it loose

Usually the approach is - if we need to change a state of an object - better we do the change as **closely as we can to the place when the state change is happening**


When I swim, I lose energy
Other birds fly - I swim - that's fine, so then when they ask me to fly - I will swim instead without even asking